projects/plugins/boost/app/modules/render-blocking-js/class-render-blocking-js.php

```diff
@@ -190,7 +190,7 @@ public function handle_output_stream( $buffer_start, $buffer_end ) {
 	 * @return array
 	 */
 	protected function get_script_tags( $buffer ) {
-		$regex = sprintf( '~<script(?![^>]*%s="%s")([^>]*)>[\s\S]*?<\/script>~si', preg_quote( $this->ignore_attribute, '~' ), preg_quote( $this->ignore_value, '~' ) );
+		$regex = sprintf( '~<script(?![^>]*%s=["\']*%s["\']*)([^>]*)>[\s\S]*?<\/script>~si', preg_quote( $this->ignore_attribute, '~' ), preg_quote( $this->ignore_value, '~' ) );
```

Comment:
Just a small suggestion to improve the accuracy. This will properly handle the ridiculous edge cases like:
```
<script data-jetpack-boost="ignore'not">console.log( "Ignore not" );</script>
```

```suggestion
		$regex = sprintf( '~<script(?![^>]*%s=(?<q>["\']*)%s\k<q>)([^>]*)>[\s\S]*?<\/script>~si', preg_quote( $this->ignore_attribute, '~' ), preg_quote( $this->ignore_value, '~' ) );
```

Reply:
Wow; I didn't know you could name capture groups; I had always done them by position which is fragile. Nice! 👍